lib/src/model/broadcast/broadcast_providers.dart

```diff
@@ -2,20 +2,48 @@ import 'package:fast_immutable_collections/fast_immutable_collections.dart';
 import 'package:lichess_mobile/src/model/broadcast/broadcast.dart';
 import 'package:lichess_mobile/src/model/broadcast/broadcast_repository.dart';
 import 'package:lichess_mobile/src/model/common/http.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part 'broadcast_providers.g.dart';
 
+/// This provider is used to get a list of tournaments broadcasts.
+/// It implements pagination with a maximum number of pages of 20.
 @riverpod
-Future<IList<Broadcast>> broadcasts(BroadcastsRef ref) async {
-  return ref
-      .withClient((client) => BroadcastRepository(client).getBroadcasts());
+class BroadcastTournamentsList extends _$BroadcastTournamentsList {
```

Comment:
I'd name it `BroadcastsPaginator`, it is more precise and then you can name the provider state `BroadcastsList` for instance.